### DISCUSSION TITLE
html wysiwyg: allow inserted images to have width and height attribute

### DISCUSSION THREAD

**Discussion by TeyKey1** [directus/directus#19566] [Feature Requests]
### Summary

Right now on image insertion in the wysiwyg editor the resulting img tag does not contain any width and height attributes. It would be nice to add an option to the wysiwyg component to insert width and height attributes when adding new images to the content.

I'd be happy to implement this if it is accepted.

### Basic Example

With this option enabled inserted image tags would look like this:
```html
<img src="https://directus-cms.com/assets/af8691eb-6c70-4558-b591-2b4600951eaa?" alt="Some alt text" width="200" height="500">
```
instead of this
```html
<img src="https://directus-cms.com/assets/af8691eb-6c70-4558-b591-2b4600951eaa?" alt="Some alt text">
```

### Motivation

If the wysiwyg content is used on a website, for example a blog page, etc. SEO aspects play a paramount role. Having images without any dimensions in the website content causes CLS issues, which in turn can heavily affect the ranking of the site.

In order to avoid CLS, the simplest solution is to add the width and height attributes to the image tag so that the browser can reserve the required space until the image is loaded. Nowadays, most modern browsers are able to infer aspect ratio of an image based on its width and height attributes, making this approach also viable for responsive pages. This approach is also considered a best-practice by [web.dev](https://web.dev/optimize-cls/#modern-best-practice-for-setting-image-dimensions).

### Detailed Design

The option to enable/disable the width and height attributes for inserted images in a wysiwyg component should not be available on a per-image basis but only once per wysiwyg component (component option). This will keep the data consistent and enforce a chosen approach among all users of the cms.

The width and height of the images will be taken from the existing implementation in [useImage.ts](https://github.com/directus/directus/blob/1857340bcdadd6d111540cc286b67cfe53c03ccd/app/src/interfaces/input-rich-text-html/useImage.ts#L159C3-L159C3) and are inserted into the tag if the component option was selected and width + height of the image are available.

### Must Have:
- An option in the wysiwyg component to enable or disable sized image tags
- Insert the correct image width and height attributes in the image tag if the option is enabled

### Should Have:
- 

### Could Have:
- 

### Won't Have:
- 

### Drawbacks

So far, no drawbacks are known to me. This can be implemented as a non-breaking change, keeping current directus defaults intact.

### Alternatives

For people who need to keep CLS on websites in check, the only alternative would be to either monkey-patch directus with this feature or create a custom extension which basically mimics the current wysiwyg image import solution of directus to add this functionality.

Alternatively (kinda hacky), one could insert the width and height attributes into the html on the frontend. I don't think this is viable though, as it would require additional API calls to get the image size and would cause overhead/longer loading times as the html needs to be processed before being displayed.

### Adoption Strategy

This can be implemented as a non-breaking change, keeping current directus defaults intact. No migration required.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
